### ISSUE TITLE
Interpolating a type causes spurious slowdown in benchmarks.

### ISSUE THREAD

**Issue by KristofferC** [JuliaCI/BenchmarkTools.jl#50]
```jl
julia> @btime round(Int, 5);
  1.539 ns (0 allocations: 0 bytes)

julia> @btime round($Int, 5);
  54.228 ns (0 allocations: 0 bytes)
```

I believe these should be equivalent.

**Comment by jrevels** [comment]
Hmm...maybe this is due to https://github.com/JuliaCI/BenchmarkTools.jl/pull/30 and is just revealing an actual performance difference in type inference?

**Comment by KristofferC** [comment]
Yeah, could be. I wonder if there is any way to work around it when benchmarking.

**Comment by tkoolen** [comment]
Has anybody found a workaround for this? I'm [calling `@benchmarkable` in a for loop in order to benchmark a bunch of `convert` methods](https://github.com/tkoolen/Rotations.jl/blob/d7c5861ff8192ae914449db38587e1c779aaed1f/perf/runbenchmarks.jl#L12), but I'm running into this issue. I'd like to avoid writing out all of the combinations by hand.

**Comment by jrevels** [comment]
I think you can just interpolate them as literals instead of arguments via `eval`, a la:

```julia
bs = Any[]
for T in (Float64, Float32, Int)
    push!(bs, eval(:(@benchmarkable convert($T, 5))))
end
```

On a side note, I'm still unsure if this should be considered a bug...it might be unexpected behavior, but AFAICT, the performance difference being shown is real. It would be incorrect for BenchmarkTools to report otherwise.

**Comment by tkoolen** [comment]
Great, that works. As long as there's a workaround, I'm happy.

**Comment by cstjean** [comment]
> the performance difference being shown is real. 

Can we build some `for` loop that exhibits this?

```julia
julia> @eval round_int(x) = round($Int, x)
round_int (generic function with 1 method)

julia> @btime round_int(5)
  1.537 ns (0 allocations: 0 bytes)
```

I just checked, and it's #30 that caused the difference. Before (faef93d0a0410b00c85afbdf863fb1bd3b468fcb) `@benchmark round($Int, 5)` is fast, and after (cf5215cf2891d32be73e1333fb5d7b89f575c76f) it's slow.

**Comment by KristofferC** [comment]
Fixed, probably by #123
